src/Framework/AspNetCoreAnalyzers/src/Analyzers/Infrastructure/ParsabilityHelper.cs

```diff
@@ -0,0 +1,161 @@
+        }
+
+        // Uri is valid.
+        if (SymbolEqualityComparer.Default.Equals(typeSymbol, wellKnownTypes.Get(WellKnownType.System_Uri)))
+        {
+            return true;
+        }
+
+        // Strings are valid.
+        if (typeSymbol.SpecialType == SpecialType.System_String)
+        {
+            return true;
+        }
+
+        return false;
+    }
+
+    internal static bool IsTypeParsable(ITypeSymbol typeSymbol, WellKnownTypes wellKnownTypes)
+    {
+        if (IsTypeAlwaysParsableOrBindable(typeSymbol, wellKnownTypes))
+        {
+            return true;
+        }
+
+        // MyType : IParsable<MyType>()
+        if (IsParsableViaIParsable(typeSymbol))
+        {
+            return true;
+        }
+
+        // Check if the parameter type has a public static TryParse method.
+        foreach (var item in typeSymbol.GetMembers("TryParse"))
+        {
+            // bool TryParse(string input, out T value)
+            if (IsTryParse(item))
+            {
+                return true;
+            }
+
+            // bool TryParse(string input, IFormatProvider provider, out T value)
+            if (IsTryParseWithFormat(item, wellKnownTypes))
+            {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    private static bool IsTryParse(ISymbol item)
+    {
+        return item is IMethodSymbol methodSymbol &&
+            methodSymbol.DeclaredAccessibility == Accessibility.Public &&
+            methodSymbol.IsStatic &&
+            methodSymbol.ReturnType.SpecialType == SpecialType.System_Boolean &&
+            methodSymbol.Parameters.Length == 2 &&
+            methodSymbol.Parameters[0].Type.SpecialType == SpecialType.System_String &&
+            methodSymbol.Parameters[1].RefKind == RefKind.Out;
+    }
+
+    private static bool IsTryParseWithFormat(ISymbol item, WellKnownTypes wellKnownTypes)
+    {
+        return item is IMethodSymbol methodSymbol &&
+            methodSymbol.DeclaredAccessibility == Accessibility.Public &&
+            methodSymbol.IsStatic &&
+            methodSymbol.ReturnType.SpecialType == SpecialType.System_Boolean &&
+            methodSymbol.Parameters.Length == 3 &&
+            methodSymbol.Parameters[0].Type.SpecialType == SpecialType.System_String &&
+            SymbolEqualityComparer.Default.Equals(methodSymbol.Parameters[1].Type, wellKnownTypes.Get(WellKnownType.System_IFormatProvider)) &&
+            methodSymbol.Parameters[2].RefKind == RefKind.Out;
+    }
+
+    private static bool IsParsableViaIParsable(ITypeSymbol typeSymbol)
+    {
+        var implementsIParsable = typeSymbol.AllInterfaces.Any(
+            i => i.Name == "IParsable" && // ... implements IParsable
+                 i.TypeArguments.Length == 1 && // ... being pedantic that it has just one type argument.
+                 i.TypeArguments.All(t => t.Name == typeSymbol.Name)); // ... and the type argument is the same as the type symbol
```

Comment:
Is there a reason for comparing via name instead of `SymbolEqualityComparer` here?